### PR TITLE
Simulator Final touches

### PR DESCRIPTION
**Pull request by jack60612**
Changes:
Make simulator look always synced, slight sim only rpc changes & a new rpc, and change the default simulator plot dir
add new simulator network mode, always make genesis block automatically in simulator, simulator is always shown as synced, @mariano54 fixed a wallet race condition caused by the wallet marking the transaction as sent and confirmed.

### CODE REVIEW

**Review by trepca — DISMISSED** [review]

**Review by mariano54 — CHANGES_REQUESTED** [review]

File: chia/consensus/network_type.py [hunk 1]
```diff
@@ -4,3 +4,4 @@
 class NetworkType(IntEnum):
     MAINNET = 0
     TESTNET = 1
+    SIMULATOR = 2
```
[the remark this answers is not in the document]
Comment (mariano54) [reply chia/consensus/network_type.py:7]: Alternatively, we can use the GENESIS_CHALLENGE

[the remark this answers is not in the document]
Comment (mariano54) [reply chia/consensus/network_type.py:7]: Or even better, `selected_network`, see this PR:
https://github.com/Chia-Network/chia-blockchain/pull/12783/files

File: chia/wallet/wallet_state_manager.py [hunk 2]
```diff
@@ -514,6 +515,9 @@ async def synced(self):
         if latest is None:
             return False
 
+        if self.constants.NETWORK_TYPE == NetworkType.SIMULATOR:
+            return True  # sim always true if we have a genesis block
```
Comment (mariano54) [inline chia/wallet/wallet_state_manager.py:519]: synced*

File: chia/consensus/network_type.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jack60612) [reply chia/consensus/network_type.py:7]: > Or even better, `selected_network`, see this PR: https://github.com/Chia-Network/chia-blockchain/pull/12783/files

I thought you said not to do that, but thats what i was going to originally do.

**Review by mariano54 — APPROVED** [review]

**Pull request merged**